hibernate-reactive-core/src/main/java/org/hibernate/reactive/stage/impl/StageSessionFactoryImpl.java

```diff
@@ -149,13 +150,17 @@ private ReactiveConnection proxyConnection(String tenantId) {
 	public <T> CompletionStage<T> withSession(Function<Stage.Session, CompletionStage<T>> work) {
 		Objects.requireNonNull( work, "parameter 'work' is required" );
 		Stage.Session current = context.get( contextKeyForSession );
-		if ( current!=null && current.isOpen() ) {
+		if ( current != null && current.isOpen() ) {
 			LOG.debug( "Reusing existing open Stage.Session which was found in the current Vert.x context" );
 			return work.apply( current );
 		}
 		else {
 			LOG.debug( "No existing open Stage.Session was found in the current Vert.x context: opening a new instance" );
-			return withSession( openSession(), work, contextKeyForSession );
+			OneOffDelegatingExecutor oneOffExecutor = new OneOffDelegatingExecutor( context );
+			final CompletionStage<T> withSessionStage = voidFuture()
+					.thenComposeAsync( v -> withSession( openSession(), work, contextKeyForSession ), oneOffExecutor );
+			oneOffExecutor.runHeldTasks();
```

Comment:
:+1:

Reply:
But is it necessary now after the other changes? Feels like you might be able to remove the `OneOffDelegatingExecutor` class, since `openSession()` isn't using _thenComposeAsync_, so this completionstages series is fully defined when it's being passed to the Executor.